### PR TITLE
doc: Improve RGW multisite documentation

### PR DESCRIPTION
**Pull request by Devp00l**
Signed-off-by: Stephan Müller <smueller@suse.com>

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/radosgw/multisite.rst [hunk 1]
```diff
@@ -503,6 +503,17 @@ example:
...
+Verification of an Object
+-------------------------
+
+By default the objects a not verified again after the synchronisation of an
+object was successful. To enable that you can set ``rgw_sync_obj_etag_verify``
+to ``true``. After you enabled the option objects that will be synchronized going
+forward will be verified by an additional MD5 checksum that is computed on the
+source and the destination, this ensures the integrity of the objects fetched
+from a remote server over HTTP including multisite sync. This option could
+decrease the performance of your RGW as more computation is needed.
```
Comment (asettle) [inline doc/radosgw/multisite.rst:515]:
```suggestion

object was successful. To enable that you can set ``rgw_sync_obj_etag_verify``to ``true``. After enabling the optional objects that will be synchronized going forward, an additional MD5 checksum will verify that it is computed on the source and the destination. This is to ensure the integrity of the objects fetched from a remote server over HTTP including multisite sync. This option can decrease the performance of your RGW as more computation is needed.

File: doc/radosgw/multisite.rst [hunk 2]
```diff
@@ -532,6 +543,25 @@ Information about the replication status of a zone can be queried with::
...
 
+The output can differ depending on the sync status. The shards are described
+as two different types during sync:
+
+- **Behind shards** are shards that need a full data sync and shards needing
+  an incremental data sync because they are not up-to-date.
+
+- **Recovery shards** are shards that encountered an error during sync and marked
+  for retry. The error mostly occurs on minor issues like acquiring a lock on
+  a bucket. So mostly nothing to worry about as it will resolve itself.
```
Comment (asettle) [inline doc/radosgw/multisite.rst:554]:
```suggestion
  a bucket. This will typically resolve itself.

File: doc/radosgw/multisite.rst [hunk 3]
```diff
@@ -532,6 +543,25 @@ Information about the replication status of a zone can be queried with::
...
+
+Check the logs
+--------------
+
+For multi site only, you can check out the metadata log (``mdlog``),
+the bucket index log (``bilog``) and the data log (``datalog``).
+You can list them and also trim them which is not needed in most cases as
+``rgw_sync_log_trim_interval`` is set to 20 minutes as default. If it isn't manually
+set to 0 you shouldn't have to trim it at any time as it could cause side effects otherwise.
```
Comment (asettle) [inline doc/radosgw/multisite.rst:563]:
```suggestion
set to 0, you shouldn't have to trim it at any time as it could cause side effects otherwise.

**Review by asettle — COMMENTED** [review]
Few nits inline :)

**Comment by Devp00l** [comment]
@asettle: Addressed all your comments :)

File: doc/radosgw/multisite.rst [hunk 4]
```diff
@@ -503,6 +503,17 @@ example:
           If the master zone is down, bucket operations executed on the
           secondary zone will fail, but object operations should succeed.
 
+Verification of an Object
+-------------------------
+
+By default the objects a not verified again after the synchronization of an
```
Comment (asettle) [inline doc/radosgw/multisite.rst:509]:
```suggestion
By default, the objects are not verified again after the synchronization of an

File: doc/radosgw/multisite.rst [hunk 5]
```diff
@@ -714,6 +744,13 @@ Multi-Site Configuration Reference
 The following sections provide additional details and command-line
 usage for realms, periods, zone groups and zones.
 
+For more details on every available configuration option, please check out
+``src/common/options/rgw.yaml.in`` or go to the more comfortable :ref:`mgr-dashboard`
+configuration page (found under `Cluster`) where you can view and set all
+options easily. On the page set the level to advanced and search for RGW,
+to see all basic and advanced configuration options with their short description.
+Expand the details of an option to reveal it's long description.
```
Comment (asettle) [inline doc/radosgw/multisite.rst:752]:
```suggestion
Expand the details of an option to reveal the long description.

File: doc/radosgw/multisite.rst [hunk 6]
```diff
@@ -714,6 +744,13 @@ Multi-Site Configuration Reference
 The following sections provide additional details and command-line
 usage for realms, periods, zone groups and zones.
 
+For more details on every available configuration option, please check out
+``src/common/options/rgw.yaml.in`` or go to the more comfortable :ref:`mgr-dashboard`
+configuration page (found under `Cluster`) where you can view and set all
+options easily. On the page set the level to advanced and search for RGW,
```
Comment (asettle) [inline doc/radosgw/multisite.rst:750]:
```suggestion
options easily. On the page, set the level to "advanced" and search for RGW,

File: doc/radosgw/multisite.rst [hunk 7]
```diff
@@ -714,6 +744,13 @@ Multi-Site Configuration Reference
...
+to see all basic and advanced configuration options with their short description.
```
Comment (asettle) [inline doc/radosgw/multisite.rst:751]:
```suggestion
to see all basic and advanced configuration options with a short description.

**Review by asettle — COMMENTED** [review]
Few nits, otherwise, LGTM

File: doc/radosgw/multisite.rst [hunk 8]
```diff
@@ -532,6 +543,25 @@ Information about the replication status of a zone can be queried with::
...
+  for retry. The error mostly occurs on minor issues like acquiring a lock on
+  a bucket. This will typically resolve itself.
+
+Check the logs
+--------------
+
+For multi site only, you can check out the metadata log (``mdlog``),
+the bucket index log (``bilog``) and the data log (``datalog``).
+You can list them and also trim them which is not needed in most cases as
+``rgw_sync_log_trim_interval`` is set to 20 minutes as default. If it isn't manually
```
Comment (tchaikov) [inline doc/radosgw/multisite.rst:562]:
```suggestion
:confval:`rgw_sync_log_trim_interval` is set to 20 minutes as default. If it isn't manually
```

**Comment by Devp00l** [comment]
Sorry for the long delay. Now I finally addressed the remaining comments.

**Review by asettle — APPROVED** [review]

File: doc/radosgw/multisite.rst [hunk 9]
```diff
@@ -503,6 +503,17 @@ example:
           If the master zone is down, bucket operations executed on the
           secondary zone will fail, but object operations should succeed.
 
+Verification of an Object
+-------------------------
+
+By default the objects a not verified again after the synchronization of an
+object was successful. To enable that you can set ``rgw_sync_obj_etag_verify``
```
Comment (tchaikov) [inline doc/radosgw/multisite.rst:510]:
```suggestion
object was successful. To enable that you can set :confval:`rgw_sync_obj_etag_verify`
```

**Comment by tchaikov** [comment]
@Devp00l none of the comment was addressed. could you please double check if you forgot to push the latest revision?

**Comment by Devp00l** [comment]
> @Devp00l not all of the comments were addressed. could you please double check if you forgot to push the latest revision?

That was really strange... I had addressed all of them again and then again... But now they are really in.

File: doc/radosgw/multisite.rst [hunk 10]
```diff
@@ -503,6 +503,17 @@ example:
           If the master zone is down, bucket operations executed on the
           secondary zone will fail, but object operations should succeed.
 
+Verification of an Object
+-------------------------
+
+By default, the objects are not verified again after the synchronization of an
+object was successful. To enable that you can set :confval:`rgw_sync_obj_etag_verify`
```
Comment (votdev) [inline doc/radosgw/multisite.rst:510]:
```suggestion
object was successful. To enable that, you can set :confval:`rgw_sync_obj_etag_verify`
```

File: doc/radosgw/multisite.rst [hunk 11]
```diff
@@ -532,6 +543,25 @@ Information about the replication status of a zone can be queried with::
...
+  an incremental data sync because they are not up-to-date.
+
+- **Recovery shards** are shards that encountered an error during sync and marked
+  for retry. The error mostly occurs on minor issues like acquiring a lock on
+  a bucket. This will typically resolve itself.
+
+Check the logs
+--------------
+
+For multi site only, you can check out the metadata log (``mdlog``),
```
Comment (votdev) [inline doc/radosgw/multisite.rst:559]: `multi-site` is used at various places in this document.
```suggestion
For multi-site only, you can check out the metadata log (``mdlog``),
```

**Comment by Devp00l** [comment]
@votdev: Addressed your comments :)

**Review by votdev — APPROVED** [review]

**Pull request merged**